common/unit_test/Test_Utils.cpp

```diff
@@ -541,3 +541,14 @@ TEST(IndexSequence, 3Dto5D) {
   EXPECT_EQ(default_axes1, ref_axes1);
   EXPECT_EQ(default_axes2, ref_axes2);
 }
+
+TEST(ToArray, lvalue) {
+  std::array<int, 3> arr{1, 2, 3};
+  ASSERT_EQ(KokkosFFT::Impl::to_array(arr), (Kokkos::Array{1, 2, 3}));
+}
+
+TEST(ToArray, rvalue) {
+  GTEST_SKIP() << "Skipping ToArray::rvalue compile time test";
+  static_assert(KokkosFFT::Impl::to_array(std::array{1, 2}) ==
```

Comment:
I am not sure we need to make it a compile time test ? We only do that for rvalue and not lvalue ?

I don't think we need to be as general as we would be inside Kokkos core, runtime tests should be enough to cover our needs. I would even consider removing the `constexpr` annotation on `to_array` ?

Reply:
I have modified the test accordingly.
As for the `constexpr` annotation, I would like to follow the `Kokkos` convention since it is likely to be an official implementation

Reply:
Ok for me